staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go

```diff
@@ -155,14 +156,23 @@ const (
 // negotiateContentEncoding returns a supported client-requested content encoding for the
 // provided request. It will return the empty string if no supported content encoding was
 // found or if response compression is disabled.
-func negotiateContentEncoding(req *http.Request) string {
+func negotiateContentEncoding(req *http.Request, disableCompressionForLocalTraffic bool) string {
 	encoding := req.Header.Get("Accept-Encoding")
 	if len(encoding) == 0 {
 		return ""
 	}
 	if !utilfeature.DefaultFeatureGate.Enabled(features.APIResponseCompression) {
 		return ""
 	}
+	if disableCompressionForLocalTraffic {
```

Comment:
Let me propose a simpler model then passing this argument around the code.

What if instead we would create a new filter.
Similarly to what we have e.g. in `WithRequestInfo` we will create a `WithDisableCompressionForLocalTraffic` filter.

And basically what it will be doing is it will be simply setting the value of the `DisableCompressionForLocalTraffic` option in the context.

That will basically give you an easy access to this value everywhere you have access to the request, including here where you need it.

Reply:
:-) @mborsz were speaking about exactly that in Slack. Definitely much simpler.

Reply:
Thanks for suggestion :)

Changed.